# P3_Réalisez une étude de santé publique

## Importer  des librairies Python et lire des bases de données

In [1164]:
#impotter des librairies
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [1165]:
#importer des database
df_population = pd.read_csv('fr_population.csv')
df_animaux = pd.read_csv('fr_animaux.csv')
df_vegetaux = pd.read_csv('fr_vegetaux.csv')
df_céréales = pd.read_csv('fr_céréales.csv')
df_sousalimentation = pd.read_csv('fr_sousalimentation.csv')

## Connaître la base de données
df_population 
df_animaux 
df_vegetaux 
df_céréales 
df_sousalimentation 

In [1166]:
df_population.columns

Index(['Code Domaine', 'Domaine', 'Code zone', 'Zone', 'Code Élément',
       'Élément', 'Code Produit', 'Produit', 'Code année', 'Année', 'Unité',
       'Valeur', 'Symbole', 'Description du Symbole'],
      dtype='object')

In [1167]:
df_population.nunique()

Code Domaine                1
Domaine                     1
Code zone                 175
Zone                      175
Code Élément                1
Élément                     1
Code Produit                1
Produit                     1
Code année                  1
Année                       1
Unité                       1
Valeur                    175
Symbole                     1
Description du Symbole      2
dtype: int64

# Question 1 : donnez le résultat de votre calcul pour l'année 2013.

In [1168]:
df_population.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 175 entries, 0 to 174
Data columns (total 14 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Code Domaine            175 non-null    object
 1   Domaine                 175 non-null    object
 2   Code zone               175 non-null    int64 
 3   Zone                    175 non-null    object
 4   Code Élément            175 non-null    int64 
 5   Élément                 175 non-null    object
 6   Code Produit            175 non-null    int64 
 7   Produit                 175 non-null    object
 8   Code année              175 non-null    int64 
 9   Année                   175 non-null    int64 
 10  Unité                   175 non-null    object
 11  Valeur                  175 non-null    int64 
 12  Symbole                 1 non-null      object
 13  Description du Symbole  175 non-null    object
dtypes: int64(6), object(8)
memory usage: 19.3+ KB


In [1169]:
df_population['Année'] .nunique()

1

Il n'y a que l'année 2013 dans la colonne année.
On peut additionner la population de chaque pays pour obtenir la population totale du monde

In [1170]:
df_population .nunique()

Code Domaine                1
Domaine                     1
Code zone                 175
Zone                      175
Code Élément                1
Élément                     1
Code Produit                1
Produit                     1
Code année                  1
Année                       1
Unité                       1
Valeur                    175
Symbole                     1
Description du Symbole      2
dtype: int64

#Il y a deux valeurs pour "Description du Symbole" voyons ce qu'elles sont :

In [1171]:
df_population ['Description du Symbole']. unique ()

array(['Donnée officielle',
       'Agrégat, peut inclure des données officielles, semi-officielles, estimées ou calculées'],
      dtype=object)

## population mondiale

In [1172]:
population_2013 = (df_population['Valeur'].sum()*1000)
population_2013

8413993000

La population de la terre est 8413993000, Mais en 2013, la population de la Terre, selon les données des Nations Unies, était de 7 210 581 976 "https://www.worldometers.info/world-population/world-population-by-year/"
)

La Chine est comptée deux fois, dans la ligne 33 les données non officielles sont en double
Si nous supprimons) la ligne contenant l'index 33 de notre ensemble de données.

In [1173]:
df_population_pas_officielles= (df_population['Description du Symbole']=='Agrégat, peut inclure des données officielles, semi-officielles, estimées ou calculées')
df_population[df_population_pas_officielles]

Code Domaine                                            Domaine  Code zone  \
33         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        351   

     Zone  Code Élément            Élément  Code Produit     Produit  \
33  Chine           511  Population totale          2501  Population   

    Code année  Année           Unité   Valeur Symbole  \
33        2013   2013  1000 personnes  1416667       A   

                               Description du Symbole  
33  Agrégat, peut inclure des données officielles,...

In [1174]:
df_population[df_population_pas_officielles]

Code Domaine                                            Domaine  Code zone  \
33         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...        351   

     Zone  Code Élément            Élément  Code Produit     Produit  \
33  Chine           511  Population totale          2501  Population   

    Code année  Année           Unité   Valeur Symbole  \
33        2013   2013  1000 personnes  1416667       A   

                               Description du Symbole  
33  Agrégat, peut inclure des données officielles,...

In [1175]:
df_population['Valeur'].sum()*1000 - 1416667*1000

6997326000

In [1176]:
#Supprimer la ligne Chine
df_population = df_population[df_population['Code zone'] != 351]

In [1177]:
df_population.loc[30:40, ['Zone','Code zone','Valeur']]

Zone  Code zone   Valeur
30                   Cameroun         32    22254
31                     Canada         33    35182
32                      Chili         40    17620
34   Chine - RAS de Hong-Kong         96     7204
35       Chine - RAS de Macao        128      566
36        Chine, continentale         41  1385567
37  Chine, Taiwan Province de        214    23330
38                     Chypre         50     1141
39                   Colombie         44    48321
40                      Congo         46     4448

In [1178]:
df_animaux = df_animaux[df_animaux['Code zone'] != 351]

In [1179]:
df_vegetaux = df_vegetaux[df_vegetaux['Code zone'] != 351]

In [1180]:
df_céréales = df_céréales[df_céréales['Code zone'] != 351]

In [1181]:
df_sousalimentation = df_sousalimentation[df_sousalimentation['Code zone'] != 351 ]

In [1182]:
df_population.shape

(174, 14)

In [1183]:
df_population ['Zone']. nunique ()

174

Note :

Il y a 206 pays dans le monde. Parmi eux, 193 sont membres des Nations Unies selon https://fr.wikipedia.org/wiki/Organisation_des_Nations_unies_pour_l%27alimentation_et_l%27agriculture, mais dans la base de données actuelle 175 

# Question 2 : Identifiez ces redondances, en donnant votre réponse sous forme de formule mathématique . 

Parmi les documents sur les Bilans alimentaires que vous avez téléchargés, il y a des informations redondantes. En effet, pour un pays donné, certaines de ces informations peuvent se calculer à partir d'autres :

Production 
Importations
Exportations 
Variation de stock 
Disponibilité intérieure 
Semences 
Pertes 
Nourriture, aussi appelée Disponibilité alimentaire 
Aliments pour animaux 
Traitement
Autres utilisations 

Identifiez ces redondances, en donnant votre réponse sous forme de formule mathématique (pas besoin de coder ici :soleil: ). C'est une équation à 3 termes de type \(a_1 + a2 + [...] = b_1 + b_2 + [...] = c_1 + c_2 + [...]\) ) faisant intervenir chacune des 11 quantités données ci dessus. Illustrez cette équation avec l'exemple du blé en France. Pour avoir un indice, 

 #Création d'un dataframe "FRANCE / BLE" pour vérifier les données

In [1184]:
df_blé_France= df_vegetaux [(df_vegetaux['Zone']== 'France') & (df_vegetaux['Produit']== 'Blé')]
df_blé_France

Code Domaine                                            Domaine  \
37487         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37488         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37489         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37490         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37491         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37492         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37493         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37494         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37495         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37496         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37497         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37498         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37499         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37500         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37501         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   

       Code zone    Zone  Code Élément  \
37487         68  France          5511   
37488         68  France          5611   
37489         68  France          5072   
37490         68  France          5911   
37491         68  France          5301   
37492         68  France          5521   
37493         68  France          5527   
37494         68  France          5123   
37495         68  France          5131   
37496         68  France          5154   
37497         68  France          5142   
37498         68  France           645   
37499         68  France           664   
37500         68  France           674   
37501         68  France           684   

                                                 Élément  Code Produit  \
37487                                         Production          2511   
37488                            Importations - Quantité          2511   
37489                                 Variation de stock          2511   
37490                            Exportations - Quantité          2511   
37491                           Disponibilité intérieure          2511   
37492                              Aliments pour animaux          2511   
37493                                           Semences          2511   
37494                                             Pertes          2511   
37495                                         Traitement          2511   
37496              Autres utilisations (non alimentaire)          2511   
37497                                         Nourriture          2511   
37498  Disponibilité alimentaire en quantité (kg/pers...          2511   
37499     Disponibilité alimentaire (Kcal/personne/jour)          2511   
37500  Disponibilité de protéines en quantité (g/pers...          2511   
37501  Disponibilité de matière grasse en quantité (g...          2511   

      Produit  Code année  Année               Unité    Valeur Symbole  \
37487     Blé        2013   2013  Milliers de tonnes  38614.00       S   
37488     Blé        2013   2013  Milliers de tonnes   2055.00       S   
37489     Blé        2013   2013  Milliers de tonnes   1131.00       S   
37490     Blé        2013   2013  Milliers de tonnes  21502.00       S   
37491     Blé        2013   2013  Milliers de tonnes  20298.00       S   
37492     Blé        2013   2013  Milliers de tonnes   7822.00       S   
37493     Blé        2013   2013  Milliers de tonnes    748.00       S   
37494     Blé        2013   2013  Milliers de tonnes    358.00       S   
37495     Blé        2013   2013  Milliers de tonnes   1575.00       S   
37496     Blé        2013   2013  Milliers de tonnes   2824.00       S   
37497     Blé        2013   2013  Milliers de tonnes   6971.00       S   
37498     Blé        2013   2013                  kg    108.43      Fc   
37499     Blé    

In [1185]:
df_blé_France.shape

(15, 14)

# Une équation à 3 termes 

Disponibilité intérieur = Production + importations - exportations + variations des stocks = Nourriture + Aliments pour animaux + Semences + Traitement + Autres utilisations + Pertes

In [1186]:
#Disponibilité intérieur
dispo_interieur1 = int(df_blé_France[df_blé_France['Code Élément'] == 5301].Valeur)
dispo_interieur1

20298

In [1187]:
#Production + importations - exportations + variations des stocks 
dispo_interieur1= (int(df_blé_France[df_blé_France['Code Élément'] == 5511].Valeur) +
                   int(df_blé_France[df_blé_France['Code Élément'] == 5611].Valeur)-
                   int(df_blé_France[df_blé_France['Code Élément'] == 5911].Valeur)+ 
                   int(df_blé_France[df_blé_France['Code Élément'] == 5072].Valeur))
dispo_interieur1

20298

In [1188]:
#Nourriture + Aliments pour animaux + Semences + Traitement + Autres utilisations + Pertes
dispo_interieur2= (int(df_blé_France[df_blé_France['Code Élément'] == 5142].Valeur) +
                   int(df_blé_France[df_blé_France['Code Élément'] == 5521].Valeur)+
                   int(df_blé_France[df_blé_France['Code Élément'] == 5527].Valeur)+
                   int(df_blé_France[df_blé_France['Code Élément'] == 5131].Valeur)+ 
                   int(df_blé_France[df_blé_France['Code Élément'] == 5154].Valeur)+ 
                   int(df_blé_France[df_blé_France['Code Élément'] == 5123].Valeur))
dispo_interieur2

20298

Ainsi, les trois termes sont exactement équivalents dans les données France/Blé, et l'équation a été vérifiée.

# Question 3 : Calculez (pour chaque pays et chaque produit) la disponibilité alimentaire en kcal puis en kg de protéines.

Vous ferez cela à partir de ces informations :

Population de chaque pays ;
Disponibilité alimentaire donnée pour chaque produit et pour chaque pays en kcal/personne/jour.
Disponibilité alimentaire en protéines donnée pour chaque produit et pour chaque pays en g/personne/jour.

In [1189]:
#Ajouter une colonne 'Radical' aux deux tables df_vegetaux et df_animaux
df_vegetaux['Radical'] = 'plante'
df_animaux['Radical'] = 'animal'

In [1190]:
df_animaux.columns

Index(['Code Domaine', 'Domaine', 'Code zone', 'Zone', 'Code Élément',
       'Élément', 'Code Produit', 'Produit', 'Code année', 'Année', 'Unité',
       'Valeur', 'Symbole', 'Description du Symbole', 'Radical'],
      dtype='object')

In [1191]:
df_vegetaux.columns

Index(['Code Domaine', 'Domaine', 'Code zone', 'Zone', 'Code Élément',
       'Élément', 'Code Produit', 'Produit', 'Code année', 'Année', 'Unité',
       'Valeur', 'Symbole', 'Description du Symbole', 'Radical'],
      dtype='object')

Les colonnes sont identiques, on peut donc fusionner les deux tables

In [1192]:
df_vegetaux_animaux = df_animaux.append (df_vegetaux) 
df_vegetaux_animaux.sample(10)

Code Domaine                                            Domaine  \
90939         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
28171         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
68834         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
31715         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
86010         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
36301         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
15535         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
10545         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
96179         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
33645         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   

       Code zone                                    Zone  Code Élément  \
90939        199                               Slovaquie          5611   
28171         54                                Danemark          5521   
68834        149                                   Népal          5142   
31715         58                                Équateur          5511   
86010        184                                  Rwanda          5511   
36301        236  Venezuela (République bolivarienne du)          5511   
15535         95                                Honduras          5911   
10545         59                                  Égypte          5511   
96179        167                           Tchéquie (la)           664   
33645        209                                Eswatini          5521   

                                              Élément  Code Produit  \
90939                         Importations - Quantité          2616   
28171                           Aliments pour animaux          2558   
68834                                      Nourriture          2518   
31715                                      Production          2745   
86010                                      Production          2517   
36301                                      Production          2766   
15535                         Exportations - Quantité          2733   
10545                                      Production          2775   
96179  Disponibilité alimentaire (Kcal/personne/jour)          2620   
33645                           Aliments pour animaux          2517   

                      Produit  Code année  Année               Unité  Valeur  \
90939       Bananes plantains        2013   2013  Milliers de tonnes     8.0   
28171  Graines Colza/Moutarde        2013   2013  Milliers de tonnes   142.0   
68834                  Sorgho        2013   2013  Milliers de tonnes     0.0   
31715                    Miel        2013   2013  Milliers de tonnes     1.0   
86010                  Millet        2013   2013  Milliers de tonnes     9.0   
36301            Cephalopodes        2013   2013  Milliers de tonnes     0.0   
15535        Viande de Suides        2013   2013  Milliers de tonnes     0.0   
10545      Plantes Aquatiques        2013   2013  Milliers de tonnes     0.0   
96179                  Raisin        2013   2013  Kcal/personne/jour     9.0   
33645                  Millet        2013   2013  Milliers de tonnes     0.0   

      Symbole Description du Symbole Radical  
90939       S  Données standardisées  plante  
28171       S  Données standardisées  plante  
68834       S  Données standardisées  plante  
31715       S  Données standardisées  plante  
86010       S  Données standardisées  plante  
36301       S  Données standardisées  animal  
15535       S  Données standardisées  animal  
10545       S  Données standardisées  animal  
96179      Fc        Donnée calculée  plante  
33645       S  Données standardisées  plante

In [1193]:
df_vegetaux_animaux.shape

(140919, 15)

In [1194]:
# Jointure avec la table population sur le Code zone 
df_vegetaux_animaux= pd.merge(df_vegetaux_animaux,df_population, on = 'Code zone' )
df_vegetaux_animaux.sample(10)

Code Domaine_x                                          Domaine_x  \
69492            FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
51334            FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
53247            FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
89253            FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
37782            FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
52247            FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
83052            FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
111201           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
103147           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
3758             FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   

        Code zone                  Zone_x  Code Élément_x  \
69492         110                   Japon            5511   
51334          75                  Gambie            5611   
53247          81                   Ghana             664   
89253         273              Monténégro             664   
37782          72                Djibouti            5301   
52247          73                 Géorgie             684   
83052         132                Maldives            5142   
111201         56  République dominicaine            5911   
103147        150                Pays-Bas             684   
3758           79               Allemagne             664   

                                                Élément_x  Code Produit_x  \
69492                                          Production            2551   
51334                             Importations - Quantité            2517   
53247      Disponibilité alimentaire (Kcal/personne/jour)            2614   
89253      Disponibilité alimentaire (Kcal/personne/jour)            2557   
37782                            Disponibilité intérieure            2560   
52247   Disponibilité de matière grasse en quantité (g...            2571   
83052                                          Nourriture            2620   
111201                            Exportations - Quantité            2764   
103147  Disponibilité de matière grasse en quantité (g...            2549   
3758       Disponibilité alimentaire (Kcal/personne/jour)            2579   

                      Produit_x  Code année_x  Année_x  ... Code Élément_y  \
69492                      Noix          2013     2013  ...            511   
51334                    Millet          2013     2013  ...            511   
53247           Agrumes, Autres          2013     2013  ...            511   
89253      Graines de tournesol          2013     2013  ...            511   
37782        Coco (Incl Coprah)          2013     2013  ...            511   
52247             Huile de Soja          2013     2013  ...            511   
83052                    Raisin          2013     2013  ...            511   
111201  Poissons Marins, Autres          2013     2013  ...            511   
103147      Légumineuses Autres          2013     2013  ...            511   
3758            Huile de Sésame          2013     2013  ...            511   

                Élément_y Code Produit_y   Produit_y Code année_y Année_y  \
69492   Population totale           2501  Population         2013    2013   
51334   Population totale           2501  Population         2013    2013   
53247   Population totale           2501  Population         2013    2013   
89253   Population totale           2501  Population         2013    2013   
37782   Population totale           2501  Population         2013    2013   
52247   Population totale           2501  Population         2013    2013   
83052   Population totale           2501  Population         2013    2013   
111201  Population totale           2501  Population         2013    2013   
103147  Population totale           2501  Population         2013    2013   
3758    Population to

In [1195]:
df_vegetaux_animaux.columns

Index(['Code Domaine_x', 'Domaine_x', 'Code zone', 'Zone_x', 'Code Élément_x',
       'Élément_x', 'Code Produit_x', 'Produit_x', 'Code année_x', 'Année_x',
       'Unité_x', 'Valeur_x', 'Symbole_x', 'Description du Symbole_x',
       'Radical', 'Code Domaine_y', 'Domaine_y', 'Zone_y', 'Code Élément_y',
       'Élément_y', 'Code Produit_y', 'Produit_y', 'Code année_y', 'Année_y',
       'Unité_y', 'Valeur_y', 'Symbole_y', 'Description du Symbole_y'],
      dtype='object')

In [1196]:
#  recréer un DataFrame avec les colonnes nécessaires
df_vegetaux_animaux= df_vegetaux_animaux[['Code Domaine_x', 'Domaine_x', 'Code zone', 'Zone_x', 'Code Élément_x',
       'Élément_x', 'Code Produit_x', 'Produit_x', 'Année_x',
       'Unité_x', 'Valeur_x','Radical', 'Valeur_y']]

In [1197]:
df_vegetaux_animaux

Code Domaine_x                                          Domaine_x  \
0                FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
1                FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
2                FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
3                FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
4                FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
...               ...                                                ...   
140914           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
140915           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
140916           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
140917           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
140918           FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   

        Code zone       Zone_x  Code Élément_x  \
0               2  Afghanistan            5511   
1               2  Afghanistan            5611   
2               2  Afghanistan            5301   
3               2  Afghanistan            5142   
4               2  Afghanistan             645   
...           ...          ...             ...   
140914        181     Zimbabwe             674   
140915        181     Zimbabwe             684   
140916        181     Zimbabwe             664   
140917        181     Zimbabwe             674   
140918        181     Zimbabwe             684   

                                                Élément_x  Code Produit_x  \
0                                              Production            2731   
1                                 Importations - Quantité            2731   
2                                Disponibilité intérieure            2731   
3                                              Nourriture            2731   
4       Disponibilité alimentaire en quantité (kg/pers...            2731   
...                                                   ...             ...   
140914  Disponibilité de protéines en quantité (g/pers...            2680   
140915  Disponibilité de matière grasse en quantité (g...            2680   
140916     Disponibilité alimentaire (Kcal/personne/jour)            2899   
140917  Disponibilité de protéines en quantité (g/pers...            2899   
140918  Disponibilité de matière grasse en quantité (g...            2899   

                    Produit_x  Année_x             Unité_x  Valeur_x Radical  \
0            Viande de Bovins     2013  Milliers de tonnes    134.00  animal   
1            Viande de Bovins     2013  Milliers de tonnes      6.00  animal   
2            Viande de Bovins     2013  Milliers de tonnes    140.00  animal   
3            Viande de Bovins     2013  Milliers de tonnes    140.00  animal   
4            Viande de Bovins     2013                  kg      4.59  animal   
...                       ...      ...                 ...       ...     ...   
140914  Aliments pour enfants     2013     g/personne/jour      0.00  plante   
140915  Aliments pour enfants     2013     g/personne/jour      0.00  plante   
140916           Miscellanees     2013  Kcal/personne/jour      2.00  plante   
140917           Miscellanees     2013     g/personne/jour      0.07  plante   
140918           Miscellanees     2013     g/personne/jour      0.03  plante   

        Valeur_y  
0          30552  
1          30552  
2          30552  
3          30552  
4          30552  
...          ...  
140914     14150  
140915     14150  
140916     14150  
140917     14150  
140918     14150  

[140919 rows x 13 columns]

In [1198]:
#Changer les noms de colonnes
df_vegetaux_animaux = df_vegetaux_animaux.rename(columns= {'Code Domaine_x':'Code Domaine', 'Domaine_x':'Domaine', 'Code zone': 'Code zone', 'Zone_x': 'Zone', 'Code Élément_x': 'Code Élément',
       'Élément_x':'Élément', 'Code Produit_x':'Code Produit', 'Produit_x': 'Produit', 'Année_x': 'Année',
       'Unité_x':'Unité', 'Valeur_x':'Valeur','Radical':'Radical', 'Valeur_y':'Population'})   

In [1199]:
df_vegetaux_animaux


Code Domaine                                            Domaine  \
0              FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
1              FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
2              FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
3              FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
4              FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
...             ...                                                ...   
140914         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
140915         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
140916         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
140917         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   
140918         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...   

        Code zone         Zone  Code Élément  \
0               2  Afghanistan          5511   
1               2  Afghanistan          5611   
2               2  Afghanistan          5301   
3               2  Afghanistan          5142   
4               2  Afghanistan           645   
...           ...          ...           ...   
140914        181     Zimbabwe           674   
140915        181     Zimbabwe           684   
140916        181     Zimbabwe           664   
140917        181     Zimbabwe           674   
140918        181     Zimbabwe           684   

                                                  Élément  Code Produit  \
0                                              Production          2731   
1                                 Importations - Quantité          2731   
2                                Disponibilité intérieure          2731   
3                                              Nourriture          2731   
4       Disponibilité alimentaire en quantité (kg/pers...          2731   
...                                                   ...           ...   
140914  Disponibilité de protéines en quantité (g/pers...          2680   
140915  Disponibilité de matière grasse en quantité (g...          2680   
140916     Disponibilité alimentaire (Kcal/personne/jour)          2899   
140917  Disponibilité de protéines en quantité (g/pers...          2899   
140918  Disponibilité de matière grasse en quantité (g...          2899   

                      Produit  Année               Unité  Valeur Radical  \
0            Viande de Bovins   2013  Milliers de tonnes  134.00  animal   
1            Viande de Bovins   2013  Milliers de tonnes    6.00  animal   
2            Viande de Bovins   2013  Milliers de tonnes  140.00  animal   
3            Viande de Bovins   2013  Milliers de tonnes  140.00  animal   
4            Viande de Bovins   2013                  kg    4.59  animal   
...                       ...    ...                 ...     ...     ...   
140914  Aliments pour enfants   2013     g/personne/jour    0.00  plante   
140915  Aliments pour enfants   2013     g/personne/jour    0.00  plante   
140916           Miscellanees   2013  Kcal/personne/jour    2.00  plante   
140917           Miscellanees   2013     g/personne/jour    0.07  plante   
140918           Miscellanees   2013     g/personne/jour    0.03  plante   

        Population  
0            30552  
1            30552  
2            30552  
3            30552  
4            30552  
...            ...  
140914       14150  
140915       14150  
140916       14150  
140917       14150  
140918       14150  

[140919 rows x 13 columns]

In [1200]:
#Population par pays
df_vegetaux_animaux['Population'] = df_vegetaux_animaux['Population'] * 1000

In [1201]:
df_vegetaux_animaux.head()

Code Domaine                                            Domaine  Code zone  \
0         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
1         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
2         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
3         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   
4         FBSH  Bilans Alimentaire (Ancienne méthodologie et p...          2   

          Zone  Code Élément  \
0  Afghanistan          5511   
1  Afghanistan          5611   
2  Afghanistan          5301   
3  Afghanistan          5142   
4  Afghanistan           645   

                                             Élément  Code Produit  \
0                                         Production          2731   
1                            Importations - Quantité          2731   
2                           Disponibilité intérieure          2731   
3                                         Nourriture          2731   
4  Disponibilité alimentaire en quantité (kg/pers...          2731   

            Produit  Année               Unité  Valeur Radical  Population  
0  Viande de Bovins   2013  Milliers de tonnes  134.00  animal    30552000  
1  Viande de Bovins   2013  Milliers de tonnes    6.00  animal    30552000  
2  Viande de Bovins   2013  Milliers de tonnes  140.00  animal    30552000  
3  Viande de Bovins   2013  Milliers de tonnes  140.00  animal    30552000  
4  Viande de Bovins   2013                  kg    4.59  animal    30552000

In [1202]:
# On créé un pivot avec les éléments en colonne
df_vegetaux_animaux = df_vegetaux_animaux.pivot_table( index = [ 'Code zone', 'Zone', 'Code Produit', 'Produit', 'Année', 'Radical', 'Population'] , columns = 'Élément', values = 'Valeur', aggfunc = sum).reset_index()

In [1203]:
df_vegetaux_animaux

Élément  Code zone     Zone  Code Produit                  Produit  Année  \
0                1  Arménie          2511                      Blé   2013   
1                1  Arménie          2513                     Orge   2013   
2                1  Arménie          2514                     Maïs   2013   
3                1  Arménie          2515                   Seigle   2013   
4                1  Arménie          2516                   Avoine   2013   
...            ...      ...           ...                      ...    ...   
15600          276   Soudan          2763      Poissons Pelagiques   2013   
15601          276   Soudan          2764  Poissons Marins, Autres   2013   
15602          276   Soudan          2805         Riz (Eq Blanchi)   2013   
15603          276   Soudan          2848       Lait - Excl Beurre   2013   
15604          276   Soudan          2899             Miscellanees   2013   

Élément Radical  Population  Aliments pour animaux  \
0        plante     2977000                   93.0   
1        plante     2977000                  137.0   
2        plante     2977000                   96.0   
3        plante     2977000                    1.0   
4        plante     2977000                    4.0   
...         ...         ...                    ...   
15600    animal    37964000                    NaN   
15601    animal    37964000                    NaN   
15602    plante    37964000                    NaN   
15603    animal    37964000                    NaN   
15604    plante    37964000                    NaN   

Élément  Autres utilisations (non alimentaire)  \
0                                          0.0   
1                                         26.0   
2                                          NaN   
3                                          NaN   
4                                          NaN   
...                                        ...   
15600                                      0.0   
15601                                      0.0   
15602                                      NaN   
15603                                      0.0   
15604                                      NaN   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  ...  \
0                                                1024.0  ...   
1                                                   0.0  ...   
2                                                   0.0  ...   
3                                                   1.0  ...   
4                                                   2.0  ...   
...                                                 ...  ...   
15600                                               0.0  ...   
15601                                               0.0  ...   
15602                                              14.0  ...   
15603                                             315.0  ...   
15604                                               7.0  ...   

Élément  Disponibilité de protéines en quantité (g/personne/jour)  \
0                                                    30.52          
1                                                     0.00          
2                                                     0.01          
3                                                     0.02          
4                                                     0.09          
...                                                    ...          
15600                                                 0.00          
15601                                                 0.04          
15602                                                 0.28          
15603                                                15.75          
15604                                                 1.47          

Élément  Disponibilité intérieure  Exportations - Quantité  \
0                           554.0                      1.0   
1                           198.0                      0.0   
2                           102.0        

In [1204]:
    # Calculs des disponibilités alimentaires totales par produit pour chaque pays (Kg) = Population * (Disponibilité de protéines en quantité (g/personne/jour))/1000 * 365
    df_vegetaux_animaux['Disponibilité alimentaire protéines totale (Kg)'] = (df_vegetaux_animaux['Population'] *df_vegetaux_animaux['Disponibilité de protéines en quantité (g/personne/jour)']/1000 * 365 )
    

In [1205]:
# Calculs des disponibilités alimentaires totales par produit pour chaque pays(kcal) = Population * (Disponibilité de protéines en quantité Disponibilité alimentaire (Kcal/personne/jour) * 365
df_vegetaux_animaux['Disponibilité alimentaire totale (Kcal)'] = (df_vegetaux_animaux['Population'] *df_vegetaux_animaux['Disponibilité alimentaire (Kcal/personne/jour)'] * 365 )

In [1206]:
df_vegetaux_animaux.sample(10)

Élément  Code zone          Zone  Code Produit                 Produit  Année  \
14862          238      Éthiopie          2657      Boissons Fermentés   2013   
6620           106        Italie          2573      Huile de Tournesol   2013   
15070          251        Zambie          2534             Racines nda   2013   
15359          272        Serbie          2555                    Soja   2013   
13021          211        Suisse          2562               Palmistes   2013   
3238            54      Danemark          2520        Céréales, Autres   2013   
2249            38     Sri Lanka          2558  Graines Colza/Moutarde   2013   
9535           149         Népal          2558  Graines Colza/Moutarde   2013   
13131          213  Turkménistan          2744                   Oeufs   2013   
14149          229   Royaume-Uni          2733        Viande de Suides   2013   

Élément Radical  Population  Aliments pour animaux  \
14862    plante    94101000                    NaN   
6620     plante    60990000                    NaN   
15070    plante    14539000                    NaN   
15359    plante     9511000                   11.0   
13021    plante     8078000                    NaN   
3238     plante     5619000                  114.0   
2249     plante    21273000                    0.0   
9535     plante    27797000                    NaN   
13131    animal     5240000                    NaN   
14149    animal    63384000                    NaN   

Élément  Autres utilisations (non alimentaire)  \
14862                                      NaN   
6620                                      90.0   
15070                                      NaN   
15359                                      4.0   
13021                                      NaN   
3238                                       NaN   
2249                                       NaN   
9535                                       NaN   
13131                                     10.0   
14149                                      NaN   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  ...  \
14862                                               7.0  ...   
6620                                              108.0  ...   
15070                                               0.0  ...   
15359                                               0.0  ...   
13021                                               NaN  ...   
3238                                                0.0  ...   
2249                                                1.0  ...   
9535                                                3.0  ...   
13131                                              30.0  ...   
14149                                             235.0  ...   

Élément  Exportations - Quantité  Importations - Quantité  Nourriture  Pertes  \
14862                        0.0                      0.0       569.0     NaN   
6620                        56.0                    293.0       275.0     NaN   
15070                        0.0                      0.0         0.0     NaN   
15359                       54.0                     16.0         0.0    20.0   
13021                        NaN                      0.0         NaN     NaN   
3238                        42.0                     49.0         0.0     0.0   
2249                         0.0                      2.0         2.0     0.0   
9535                         0.0                     76.0         6.0     4.0   
13131                        NaN                      2.0        41.0     NaN   
14149                      246.0                   1047.0      1635.0     NaN   

Élément  Production  Semences  Traitement  Variation de stock  \
14862         569.0       NaN         NaN                 0.0   
6620          129.0       NaN         0.0                 0.0   
15070           NaN       NaN         NaN                 0.0   
15359         385.0       6.0       230.0               -75.0   
13021           NaN       NaN         0.

#Résultat : Dans le monde, l'accès à la nourriture en quantité n'est pas égalitaire

# Question 4 : Ratios énergétiques
A partir de ces dernières informations, et à partir du poids de la disponibilité alimentaire (pour chaque pays et chaque produit), calculez pour chaque produit le ratio "énergie/poids", que vous donnerez en kcal/kg. Vous pouvez vérifier la cohérence de votre calcul en comparant ce ratio aux données disponibles sur internet, par exemple en cherchant la valeur calorique d'un oeuf.

Indication : La disponibilité alimentaire en kcal/personne/jour est calculée par la FAO en multipliant la quantité Nourriture par le ratio énergie/poids (en kcal/kg), puis en le divisant par la population du pays puis par 365. Ici, on vous demande juste de retrouver le ratio énergie/poids que la FAO a utilisé dans son calcul.

In [1240]:
df_vegetaux_animaux['Nourriture'].replace(0, np.nan, inplace=True)

In [1241]:
#Calcul du ratio énergie/poids (en kcal/kg)= La disponibilité alimentaire en kcal/kg / la quantité Nourriture
df_vegetaux_animaux['Ratio énergie/poids (kcal/kg)'] = (df_vegetaux_animaux['Disponibilité alimentaire totale (Kcal)']
                                                        / (df_vegetaux_animaux['Nourriture'] * 1000000)) 


In [1242]:
df_vegetaux_animaux_Oeufs = df_vegetaux_animaux[df_vegetaux_animaux['Produit'] == 'Oeufs']
df_vegetaux_animaux_Oeufs 

Élément  Code zone         Zone  Code Produit Produit  Année Radical  \
74               1      Arménie          2744   Oeufs   2013  animal   
144              2  Afghanistan          2744   Oeufs   2013  animal   
226              3      Albanie          2744   Oeufs   2013  animal   
319              4      Algérie          2744   Oeufs   2013  animal   
400              7       Angola          2744   Oeufs   2013  animal   
...            ...          ...           ...     ...    ...     ...   
15231          255     Belgique          2744   Oeufs   2013  animal   
15324          256   Luxembourg          2744   Oeufs   2013  animal   
15416          272       Serbie          2744   Oeufs   2013  animal   
15506          273   Monténégro          2744   Oeufs   2013  animal   
15596          276       Soudan          2744   Oeufs   2013  animal   

Élément  Population  Aliments pour animaux  \
74          2977000                    NaN   
144        30552000                    NaN   
226         3173000                    NaN   
319        39208000                    NaN   
400        21472000                    NaN   
...             ...                    ...   
15231      11104000                    NaN   
15324        530000                    NaN   
15416       9511000                    NaN   
15506        621000                    NaN   
15596      37964000                    NaN   

Élément  Autres utilisations (non alimentaire)  \
74                                         0.0   
144                                        NaN   
226                                        NaN   
319                                        NaN   
400                                        NaN   
...                                        ...   
15231                                      NaN   
15324                                      NaN   
15416                                      NaN   
15506                                      NaN   
15596                                      NaN   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  ...  Nourriture  \
74                                                 45.0  ...        35.0   
144                                                 5.0  ...        40.0   
226                                                46.0  ...        40.0   
319                                                27.0  ...       316.0   
400                                                 4.0  ...        24.0   
...                                                 ...  ...         ...   
15231                                              49.0  ...       140.0   
15324                                              55.0  ...         8.0   
15416                                              34.0  ...        85.0   
15506                                              44.0  ...         7.0   
15596                                               4.0  ...        38.0   

Élément  Pertes  Production  Semences  Traitement  Variation de stock  \
74          NaN        34.0       0.0         NaN                 NaN   
144         5.0        18.0       2.0         NaN                 NaN   
226         7.0        48.0       0.0         NaN                 NaN   
319        10.0       347.0      21.0         NaN                 NaN   
400         2.0         5.0       3.0         NaN                 NaN   
...         ...         ...       ...         ...                 ...   
15231       5.0       174.0      18.0         NaN                 NaN   
15324       NaN         2.0       0.0         NaN                 0.0   
15416       NaN        88.0       4.0         NaN                 NaN   
15506       NaN         4.0       0.0         NaN                 NaN   
15596       NaN        40.0       4.0         NaN                 NaN   

Élément  Disponibilité alimentaire protéines totale (Kg)  \
74                                            3727055.15   
144                                           4237562.40   
226        

Les calories dans les œufs (155 kcal) en 100 g, soit 1550 par kg, selon Wikipedia https://fr.wikipedia.org/wiki/%C5%92uf_(aliment) 
Il est curieux de voir à quel point les ratios énergétiques diffèrent de la valeur attendue, mais on constate que la valeur moyenne est proche (à 1345 kcal/kg).

In [1272]:

df_vegetaux_animaux.describe()

Élément     Code zone  Code Produit    Année    Population  \
count    15605.000000  15605.000000  15605.0  1.560500e+04   
mean       125.334508   2631.376610   2013.0  4.219567e+07   
std         73.151148     91.753174      0.0  1.502135e+08   
min          1.000000   2511.000000   2013.0  5.400000e+04   
25%         63.000000   2558.000000   2013.0  2.839000e+06   
50%        120.000000   2613.000000   2013.0  9.511000e+06   
75%        188.000000   2732.000000   2013.0  2.893400e+07   
max        276.000000   2899.000000   2013.0  1.385567e+09   

Élément  Aliments pour animaux  Autres utilisations (non alimentaire)  \
count              2720.000000                            5496.000000   
mean                479.501838                             157.391376   
std                4240.119637                            5076.785816   
min                   0.000000                               0.000000   
25%                   0.000000                               0.000000   
50%                   4.000000                               0.000000   
75%                  74.000000                               4.000000   
max              150000.000000                          347309.000000   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  \
count                                      14241.000000   
mean                                          34.789832   
std                                          107.287655   
min                                          -21.000000   
25%                                            0.000000   
50%                                            4.000000   
75%                                           21.000000   
max                                         1711.000000   

Élément  Disponibilité alimentaire en quantité (kg/personne/an)  \
count                                         14015.000000        
mean                                              8.719368        
std                                              24.618223        
min                                              -1.930000        
25%                                               0.060000        
50%                                               0.830000        
75%                                               5.190000        
max                                             430.760000        

Élément  Disponibilité de matière grasse en quantité (g/personne/jour)  \
count                                         11794.000000               
mean                                              1.283111               
std                                               3.680399               
min                                              -0.030000               
25%                                               0.010000               
50%                                               0.080000               
75%                                               0.630000               
max                                              60.760000               

Élément  Disponibilité de protéines en quantité (g/personne/jour)  ...  \
count                                         11561.000000         ...   
mean                                              1.223608         ...   
std                                               3.598686         ...   
min                                              -0.370000         ...   
25%                                               0.010000         ...   
50%                                               0.100000         ...   
75%                                               0.660000         ...   
max                                              54.970000         ...   

Élément     Nourriture        Pertes     Production      Semences  \
count      9997.000000   4278.000000    9180.000000   2091.000000   
mean        487.772132    106.053763    1090.379085     73.974653   
std        5292.991932   1113.100416   12067.344094    528.069224   
min        -246.000000      0.000000 

In [1246]:
#le pourcentage de protéines de chaque produit (pour chaque pays).
df_vegetaux_animaux['Pourcentage de protéines produit'] = (df_vegetaux_animaux['Disponibilité alimentaire protéines totale (Kg)']  / (df_vegetaux_animaux['Nourriture'] * 1000000))*100

In [1274]:
df_vegetaux_animaux[df_vegetaux_animaux['Produit'] == 'Avoine'].describe()

Élément   Code zone  Code Produit   Année    Population  \
count    167.000000         167.0   167.0  1.670000e+02   
mean     125.269461        2516.0  2013.0  4.097872e+07   
std       72.337265           0.0     0.0  1.484793e+08   
min        1.000000        2516.0  2013.0  5.400000e+04   
25%       66.500000        2516.0  2013.0  2.811500e+06   
50%      120.000000        2516.0  2013.0  9.357000e+06   
75%      184.500000        2516.0  2013.0  2.831300e+07   
max      276.000000        2516.0  2013.0  1.385567e+09   

Élément  Aliments pour animaux  Autres utilisations (non alimentaire)  \
count               122.000000                              13.000000   
mean                133.204918                              15.846154   
std                 400.507672                              50.875413   
min                   0.000000                               0.000000   
25%                   0.000000                               0.000000   
50%                   2.500000                               1.000000   
75%                  83.000000                               4.000000   
max                3298.000000                             185.000000   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  \
count                                        158.000000   
mean                                           5.569620   
std                                            9.643563   
min                                          -21.000000   
25%                                            0.000000   
50%                                            1.000000   
75%                                            8.000000   
max                                           60.000000   

Élément  Disponibilité alimentaire en quantité (kg/personne/an)  \
count                                           157.000000        
mean                                              0.964522        
std                                               1.598133        
min                                              -1.930000        
25%                                               0.020000        
50%                                               0.220000        
75%                                               1.370000        
max                                               9.490000        

Élément  Disponibilité de matière grasse en quantité (g/personne/jour)  \
count                                           141.000000               
mean                                              0.108440               
std                                               0.173873               
min                                              -0.030000               
25%                                               0.000000               
50%                                               0.030000               
75%                                               0.150000               
max                                               1.120000               

Élément  Disponibilité de protéines en quantité (g/personne/jour)  ...  \
count                                           150.000000         ...   
mean                                              0.238000         ...   
std                                               0.384988         ...   
min                                              -0.370000         ...   
25%                                               0.010000         ...   
50%                                               0.065000         ...   
75%                                               0.320000         ...   
max                                               2.320000         ...   

Élément   Nourriture     Pertes   Production     Semences  Traitement  \
count      94.000000  65.000000    76.000000    71.000000    7.000000   
mean       41.521277   7.907692   313.368421    35.000000    5.142857   
std       148.970787  14.977820   754.603134   155.261991   13.171036   
min      -246.000000   0.000000     0.000000    

Le pourcentage de protéines dans l'avoine est Protéine.
16,9 g selon https://en.wikipedia.org/wiki/Oat#Nutrient_profile

# Question 5 :Citez 5 aliments parmi les 20 aliments les plus riches en protéines.
    Citez 5 aliments parmi les 20 aliments les plus caloriques, en utilisant le ratio énergie/poids.
Étonnamment, il arrive que ce ratio soit différent en fonction du pays. Il faudra donc réaliser pour chaque aliment une moyenne sur les différents pays. Vous créerez donc une nouvelle table grâce à une agrégation. Attention à bien retirer les valeurs égales à 0 afin de ne pas fausser le calcul de la moyenne.




In [1302]:
# Créer un nouveau DataFrame avec  les valaurs supérieur à 0

df_grandes_calorique = df_vegetaux_animaux[df_vegetaux_animaux['Ratio énergie/poids (kcal/kg)'] > 0]



In [1303]:
# Puis on applique un group by sur le Produit 
df_grandes_calorique = df_grandes_calorique.groupby('Produit').mean().reset_index()


In [1304]:
df_grandes_calorique

Élément                   Produit   Code zone  Code Produit   Année  \
0                Abats Comestible  127.664596        2736.0  2013.0   
1                 Agrumes, Autres  132.459459        2614.0  2013.0   
2           Aliments pour enfants  125.843373        2680.0  2013.0   
3                          Ananas  121.652893        2618.0  2013.0   
4        Animaux Aquatiques Autre  117.500000        2769.0  2013.0   
..                            ...         ...           ...     ...   
89               Viande de Suides  123.979866        2733.0  2013.0   
90            Viande de Volailles  125.431034        2734.0  2013.0   
91                  Viande, Autre  133.812500        2735.0  2013.0   
92                            Vin  121.000000        2655.0  2013.0   
93                 Épices, Autres  127.672566        2645.0  2013.0   

Élément    Population  Aliments pour animaux  \
0        4.344527e+07              45.863636   
1        6.133273e+07                    NaN   
2        3.488387e+07                    NaN   
3        4.919470e+07                    NaN   
4        3.772612e+08                    NaN   
..                ...                    ...   
89       4.044692e+07                    NaN   
90       4.021452e+07               0.000000   
91       3.741677e+07               5.250000   
92       3.164736e+07                    NaN   
93       5.772540e+07                    NaN   

Élément  Autres utilisations (non alimentaire)  \
0                                    16.307692   
1                                     3.000000   
2                                          NaN   
3                                     0.428571   
4                                     0.000000   
..                                         ...   
89                                    8.090909   
90                                   31.450000   
91                                    1.363636   
92                                   72.952381   
93                                   20.111111   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  \
0                                              8.894410   
1                                              3.351351   
2                                              6.024096   
3                                              5.371901   
4                                              1.000000   
..                                                  ...   
89                                           102.369128   
90                                            74.879310   
91                                             9.098214   
92                                            17.356522   
93                                             7.159292   

Élément  Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                                 2.853106        
1                                                 4.808108        
2                                                 0.593614        
3                                                 5.281405        
4                                                 0.747500        
..                                                     ...        
89                                               16.576913        
90                                               20.561322        
91                                                2.688482        
92                                                9.069913        
93                                                0.756106        

Élément  Disponibilité de matière grasse en quantité (g/personne/jour)  ...  \
0                                                 0.264099              ...   
1                                                 0.022941              ...   
2                                                 0.047470              ...   
3                                                 0.017273              ...   
4                                                 0.007500  

In [1305]:
df_grandes_calorique.sort_values(by=['Ratio énergie/poids (kcal/kg)'], inplace=True, ascending=False)

In [1307]:
df_grandes_calorique.head(20)

Élément                   Produit   Code zone  Code Produit   Année  \
43       Huiles de Foie de Poisso  193.000000        2782.0  2013.0   
41                Huile de Sésame  119.200000        2579.0  2013.0   
40            Huile de Son de Riz  119.166667        2581.0  2013.0   
44             Huiles de Poissons   77.333333        2781.0  2013.0   
32               Huile d'Arachide  129.328947        2572.0  2013.0   
33                  Huile d'Olive  126.235294        2580.0  2013.0   
36         Huile de Germe de Maïs  131.269231        2582.0  2013.0   
30        Huil Plantes Oleif Autr  131.041667        2586.0  2013.0   
38             Huile de Palmistes  138.346939        2576.0  2013.0   
31         Huile Graines de Coton  130.555556        2575.0  2013.0   
37                 Huile de Palme  128.514286        2577.0  2013.0   
42             Huile de Tournesol  128.685484        2573.0  2013.0   
39                  Huile de Soja  124.940299        2571.0  2013.0   
35        Huile de Colza&Moutarde  128.348485        2574.0  2013.0   
34                  Huile de Coco  135.397260        2578.0  2013.0   
28         Graisses Animales Crue  127.882759        2737.0  2013.0   
9                    Beurre, Ghee  126.968992        2740.0  2013.0   
60                      Palmistes   39.000000        2562.0  2013.0   
84                         Sésame  129.898551        2561.0  2013.0   
5          Arachides Decortiquees  129.220588        2556.0  2013.0   

Élément    Population  Aliments pour animaux  \
43       2.041050e+07                    NaN   
41       1.173457e+08                    NaN   
40       2.863371e+08                    NaN   
44       6.568467e+07                  21.50   
32       7.243880e+07                    NaN   
33       4.734812e+07                   0.00   
36       5.508121e+07                    NaN   
30       4.624730e+07                   0.00   
38       7.860473e+07                    NaN   
31       8.473535e+07                    NaN   
37       5.593403e+07                   0.00   
42       4.956136e+07                    NaN   
39       5.022874e+07                   2.75   
35       7.668285e+07                 700.00   
34       7.583960e+07                    NaN   
28       4.798956e+07                 155.60   
9        5.197145e+07                   1.00   
60       1.335100e+07                    NaN   
84       7.676384e+07                    NaN   
5        4.733350e+07                   1.00   

Élément  Autres utilisations (non alimentaire)  \
43                                    0.000000   
41                                   57.200000   
40                                   66.000000   
44                                    4.000000   
32                                   47.916667   
33                                    6.098039   
36                                    6.727273   
30                                   29.554622   
38                                  135.333333   
31                                   33.419355   
37                                  277.171429   
42                                   46.070423   
39                                  245.915493   
35                                  278.272727   
34                                   20.230769   
28                                   76.892857   
9                                     1.400000   
60                                         NaN   
84                                    6.333333   
5                                    70.277778   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  \
43                                            10.000000   
41                                             9.314286   
40                                             7.916667   
44                                            34.000000   
32                                            26.000000   
33                                            28.188235   
36                                 